packages/entity-list/src/modules/list/sagas.js

```diff
@@ -166,15 +166,30 @@ export function* fetchEntitiesAndAddToStore(page) {
 }
 
 export function* requestEntities(page) {
-  const list = yield select(listSelector)
-  const {entityStore} = list
+  const {entityStore} = yield select(listSelector)
 
   if (!entityStore[page]) {
     yield call(fetchEntitiesAndAddToStore, page)
   }
 
   yield call(displayEntity, page)
-  yield spawn(fetchEntitiesAndAddToStore, page + 1)
+  yield spawn(preloadNextPage, page)
+}
+
+export function* preloadNextPage(currentPage) {
+  const list = yield select(listSelector)
+  const {entityStore, limit} = list
+  let {entityCount} = list
+  const nextPage = currentPage + 1
+
+  if (entityCount === null) {
```

Comment:
what about undefined? maybe this should be considered as well to make that line a little bit more stable?